openedx/features/course_experience/__init__.py

```diff
@@ -12,11 +13,88 @@
 # Namespace for course experience waffle flags.
 WAFFLE_FLAG_NAMESPACE = WaffleFlagNamespace(name='course_experience')
 
+
+class DefaultTrueWaffleFlagNamespace(WaffleFlagNamespace):
+    """
+    This is a temporary class to help deprecate/remove ``flag_undefined_default``.
+
+    TODO: TNL-7061: Perform the actual clean-up required to remove these flags
+        and refactor/fix any tests that shouldn't be removed.
+
+    """
+    def _is_flag_active(self, flag_name):
+        """
+        Returns and caches whether the provided flag is active.
+
+        If the flag value is already cached in the request, it is returned.
+        If the flag doesn't exist, always returns default of True.
+        """
+        # Import is placed here to avoid model import at project startup.
+        from waffle.models import Flag
+
+        # validate arguments
+        namespaced_flag_name = self._namespaced_name(flag_name)
+        default_true_cache_name = 'default_true.{}'.format(namespaced_flag_name)
+
+        value = self._cached_flags.get(default_true_cache_name)
+        if value is None:
+
+            # determine if the flag is undefined in waffle
+            try:
+                Flag.objects.get(name=namespaced_flag_name)
+            except Flag.DoesNotExist:
+                # default to True if not defined
+                value = True
+
+            if value is None:
+                request = crum.get_current_request()
+                if request:
+                    value = flag_is_active(request, namespaced_flag_name)
+                else:
+                    # Return the default value if not in a request context.
+                    # Same as the original implementation
+                    return True
+
+            self._cached_flags[default_true_cache_name] = value
+
+        return value
+
+    def is_flag_active(self, flag_name, check_before_waffle_callback=None, flag_undefined_default=None):
+        """
+        Overrides is_flag_active.
+
+        Phase 1 (this version): Compares a new implementation with the original implementation to ensure
+        the values aren't changing in Production.
+        Phase 2: This will simply return the new value.
+
+        """
+        default_true_value = self._is_flag_active(flag_name)
+        original_value = super().is_flag_active(flag_name, check_before_waffle_callback, flag_undefined_default=True)
+
+        is_match = (default_true_value == original_value)
+        set_custom_metric('temp_default_true_value_matches', is_match)
+        if not is_match:
+            # TODO: REMOVE THIS BOMB BEFORE MERGING!!!!
+            # It will just show we get the same value during all tests.
+            raise Exception('Boom!')
```

Comment:
**IMPORTANT: DO NOT MERGE WITH THIS HERE. :)**  I want this here for the first round of unit testing.